### ISSUE TITLE
H2 minimal basis set FCI yields NaN

### ISSUE THREAD

**Issue by susilehtola** [psi4/psi4#561]
```
molecule {
symmetry c1
H 0.0 0.0 0.0
H 0.0 0.0 1000.0
}

set basis STO-3G
energy('fci')
```

gives out NaNs after the first iteration.

**Comment by dgasmith** [comment]
Sure, we get a NaN because an occupied and a virtual eigenvalue have the exact same energy. Its an unphysical reference that we didn't anticipate. We probably wont patch this as its an extreme edge case, you can fix it for yourself by `set HD_AVG HD_KAVE`.

**Comment by susilehtola** [comment]
With your suggestion of set HD_AVG HD_KAVE
```
*** H0 Block Eigenvalue =  -0.93316370

Find the roots by the Simultaneous Expansion Method (Block Davidson Method)
Using 1 initial trial vectors
(sem_iter): H0block_->H0b_diag[0][0] - H0block_->H0b_diag[1][0] = 1.000000 - 0.000000 = 1.000000 > 1.0E-8
(sem_iter): H0block_->H0b_diag[0][1] - H0block_->H0b_diag[1][1] = 0.000000 - 1.000000 = -1.000000 > 1.0E-8
Iter  0  Root  1 =  -0.159086932   Delta_E -1.596E-01   Delta_C  0.000E+00  
Warning: Norm of correction (root 0) is < 1.0E-13
Iter  1  Root  1 =           nan   Delta_E        NAN   Delta_C       -NAN  
Warning: Norm of correction (root 0) is < 1.0E-13
Iter  2  Root  1 =           nan   Delta_E        NAN   Delta_C        NAN  
```

I'm not sure I understand why the H0 eigenvalue is close to -1, but the 0th iteration has a very bad value...

**Comment by dgasmith** [comment]
Yea, I realized that `HD_KAVE` only delays this result to about 50 angstroms, so I had removed that suggestion.

The FCI Hamiltonian of your supplied geometry is diagonal so your eigenvectors are unit vectors (hence the `1`'s). The warnings are, in perhaps an obscure way, telling you that the wrong symmetry is found in the H0 block eigenvectors. The overlap between the Gaussian basis functions is a hard zero after about 45 angstroms, so no coupling can occur.

For what its worth MOLPRO returns the same result.
